kernelci.org/content/en/docs/instances/local.md

```diff
@@ -0,0 +1,233 @@
+
+1. Host machine which will be used to connect to the VM
+   It needs to meet following requirements:
+    - Installed apps:
+      - _git_
+      - _ansible_
+      - _Python 2.7_
+      - _Python 3.6+_
+
+## Deploy KernelCI backend and storage
+
+> # Note
+> It is assumed in this section that your kernelci VM is available from your host with the hostname `kci-vm`
+
+1. Make sure _ansible_ is installed on the host machine.
+2. Clone `kernelci-backend-config` repository
+
+```bash
+git clone git@github.com:kernelci/kernelci-backend-config.git
+```
+
+3. Configure ansible to be used with your VM
+
+    ```bash
+    cd kernelci-backend-config
+    ```
+
+    - Modify `[local]` section of the `hosts` file to match your configuration
+
+    ```ini
+    [local]
+    kci-vm ansible_ssh_user=kci
+    ```
+
+    - Create `host_vars/kci-vm`
+
+    ```bash
+    touch host_vars/kci-vm
+    ```
+
+    - Edit `host_vars/kci-vm` with your favorite text editor and put content there:
+
+    ```
+    hostname: kci-vm 
+    role: production
+    certname: kci-vm
+    storage_certname: kci-vm
+    kci_storage_fqdn: kci-vm
+    become_method: su
+    ```
+
+    - Create `dev.yml` playbook file with the following content:
+
+    ```
+    - hosts: kci-vm
+        become: yes
+        become_method: su
+        become_user: root
+        gather_facts: yes
+        roles:
+            - common
+            - install-deps
+            - install-app
+            - init-conf
+            - kernelci-storage
+
+    ```
+
+    - Prepare secrets
+      - Copy `secrets.yml` template
+
+        ```
+        cp templates/secrets.yml dev-secrets.yml
+        ```
+
+      - Fill out necessary values
+
+        ```
+        master_key:
+            "<random_string>"
+
+        # The url location where the backend will be running
+        backend_url:
+            "http://kci-vm"
+        # The url location where the frontend will be running
+        base_url:
+            "http://kci-vm"
+        # The backend token OR master key for a fresh install
+        # If set to the master-key this field will have to be updated
+        # once the tokens created
+        backend_token:
+            "<random_string>"
+        # A secret key internally used by Flask
+        secret_key:
+            "<random_string>"
+        # The url location of the storage server for the backend
+        file_server:
+            "http://kci-vm"
+        ```
+
+        ```
+        ssl_stapling_resolver:
+            "localhost"
+        ```
+
+
+> You can use UUID as your random string. You can easily generate one with
+> ```
+> python -c "import uuid;print(uuid.uuid4())"
+> ```
+
+   - Run ansible playbook to set up your VM
+
+      ```
+      ansible-playbook -i hosts -l kci-vm -D -b -K  -e git_head="main" -e "@./dev-secrets.yml" --key-file ~/.ssh/org/id_rsa dev.yml
+      ```
+
+> Note
+> The nginx configuration step may fail due to SSL issues, but that's fine for the development setup.
```

Comment:
Maybe this can be fixed with some conditional in the Ansible playbook?

Reply:
Actually, I have an idea for a fix, but I'd like to test it first and then got merged to `kernelci-backend-config`

Reply:
> Looks good, couple of suggestions for debugging purpose for someone new to ansible:
> 
> After updating `dev.yml` validate the file with command:
> 
> ```
> ansible-playbook dev.yml --syntax-check
> ```
> 
> Remember to enable verbose logs if you encounter issue with ansible-playbook like
> 
> ```
> ansible-playbook -vvv
> ```

Thanks, I'll add notes to the instructions